### ISSUE TITLE
Add `at-rule-descriptor-no-unknown`

### ISSUE THREAD

**Issue by jeddy3** [stylelint/stylelint#8148]
### What is the problem you're trying to solve?

A dedicated rule to flag unknown descriptors within at-rules.

For example, `margin` is not a known descriptor for `@counter-style`:

```css
@counter-style foo { margin: 0; }
```

### What solution would you like to see?

Suggested blueprint:

- Name: `at-rule-descriptor-no-unknown`
- Description: Disallow unknown descriptors for at-rules.
- Message: `Unexpected unknown descriptor "${descriptor}" for at-rule "${at-rule}"`
- Primary option: `true`
- Secondary options: n/a
- Category: Avoid errors > Unknown
- Included in sharable configs?: Yes, recommended.
- Expanded description:

This rule considers values for properties defined within the CSS specifications to be known.

Certain at-rules only allow specific descriptors. This rule checks that. For example, only the `syntax`, `inherits` and 
`initial-value` descriptors are known to the `@property` at-rule.

**Comment by ybiquitous** [comment]
This idea sounds good to me 👍🏼 

But I have a concern only about the naming consistency. We already have the [`at-rule-property-required-list`](https://github.com/stylelint/stylelint/blob/ca8d5b429dadd0e0c401dd316d9e0d7da9d5cf7b/lib/rules/at-rule-property-required-list/README.md) rule, but the new rule would start with `at-rule-descriptor`. Are there some differences between `at-rule-property` and `at-rule-descriptor`?

**Comment by jeddy3** [comment]
Yes, they are [different things](https://drafts.csswg.org/css-syntax/#declaration):

> Declarations are further categorized as property declarations or descriptor declarations, with the former setting CSS [properties](https://drafts.csswg.org/css-cascade-5/#css-property) and appearing most often in [qualified rules](https://drafts.csswg.org/css-syntax/#qualified-rule) and the latter setting CSS [descriptors](https://drafts.csswg.org/css-syntax/#css-descriptor), which appear only in [at-rules](https://drafts.csswg.org/css-syntax/#at-rule).

In hindsight, we probably should've named the `at-rule-property-required-list` rule `at-rule-descriptor-required-list`. I think we're fine with leaving it as it is to minimise disruption to users. Perhaps we can tweak the rule description to mention descriptors:

> Specify a list of required properties for an at-rule.

Becomes "Specify a list of required descriptors (previously labelled properties) for an at-rule."

Going forward, I think we should use the correct term of descriptors so that we avoid issues like https://github.com/stylelint/stylelint/issues/8150#issuecomment-2531029925.

**Comment by ybiquitous** [comment]
Thanks for the detailed explanation! I now understand the difference between properties and descriptors deeply. 👍🏼 

The CSS spec you quoted also says:

> This categorization does not occur at the Syntax level; instead, it is a product of where the declaration appears, and is defined by the respective specifications defining the given rule.

That's why parsers, which we're using now (e.g., PostCSS, CSSTree, etc.), don't distinguish between them. Both are equal at the syntax level.

In short, I think the `at-rule-descriptor-no-unknown` rule name also makes sense. 👍🏼 

Besides, I will open a document fixing PR for `at-rule-property-required-list` soon, as you suggested.
